Improved docs on `map` matching.

nisbet-hubbard opened the pull request:
This PR improves the wording regarding `map` testing to reflect the performance benefits of using exact match over regex.

https://web.archive.org/web/20240908025132/https://forum.nginx.org/read.php?2,214693,214694

y82 commented:
Hi @nisbet-hubbard,

Thank you for your contribution and for your attention to detail. I still find the existing phrasing — “regular expression match” and “matching variant chosen” — quite clear and consistent with our current terminology. Additionally, we haven’t previously used the term “testing” in this context, while phrases like “strings evaluated/matched” and “regular expressions matched” are more established.

This is just my perspective, and other community members may have different thoughts on this.

pluknet commented:
In my personal reading of the current wording:
"the first matching variant will be chosen"
already implies that
"the search terminates on the first match".

Further, regarding the patch, I don't like that "the first matching" is used twice, it worsens the wording quality.

pluknet committed:
Refined description of map and server names matching.

The search is performed up to the first match.
The search priority is always applied.

While here, simplified description.
Unified wording based on text review from f09e5433.

pluknet commented:
Hello,

I've looked once again at the current state.
The text reflects the implementation detail for ngx_hash_find_combined() based search.
So it needs to be updated in other places such as searching server names.
This also need to include the "stream" counterpart.

Regardless, the existing wording looks like it has a logic flaw.
Personally, I read it as if the priority is applied only for the case of multiple search candidates.

Since it's hard to explain in details, and to reduces the number of review round-trips,
I have updated your branch to address all the details (including translations).
Please see if it's good for you.  I intend to merge this in a couple days.
